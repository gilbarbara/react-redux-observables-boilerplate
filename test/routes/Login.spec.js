import React from 'react';
import { mount } from 'enzyme';

import Login from 'routes/Login';

function setup() {
  const props = {
    location: {},
  };

  return mount(<Login {...props} />);
}

describe('Login', () => {
  const wrapper = setup(true);

  it('should be a StatelessComponent', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the redirect location', () => {
    wrapper.setProps({
      location: {
        state: { from: '/private' },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
