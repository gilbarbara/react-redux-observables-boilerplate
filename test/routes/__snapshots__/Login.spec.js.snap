// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render properly 1`] = `"<div class=\\"app__login app__route\\"><div class=\\"app__container\\"><h1>You must login to view this page</h1></div></div>"`;

exports[`Login should render the redirect location 1`] = `"<div class=\\"app__login app__route\\"><div class=\\"app__container\\"><h1>You must login to view /private</h1></div></div>"`;
